### ISSUE TITLE
[HOLD for payment 2025-02-05] [$250] Tweak UI for deleted message and expense

### ISSUE THREAD

**Issue by dubielzyk-expensify** [Expensify/App#53617]
## Problem

Today when you delete a message or expense with threads, the chat message gets replaced with `[Deleted message]` or `[Deleted expense]`. This feels a bit unpolished:

![CleanShot 2024-12-05 at 13 59 52@2x](https://github.com/user-attachments/assets/ab04b324-c1f1-49f6-8584-22ed6f511e79)

## Solution

Let's spruce up the UI and reuse our attachement when showing a deleted message or expense:

![CleanShot 2024-12-09 at 10 24 35@2x](https://github.com/user-attachments/assets/7680384f-aa4f-4755-abbb-59fbeea129b3)

For the receipt deleted we should use the icon already in our repo (`App/assets/images/receipt-slash.svg`). For the comment deleted, we should use this new icon:
[chatbubble-slash.svg.zip](https://github.com/user-attachments/files/18054701/chatbubble-slash.svg.zip)

cc @Expensify/design @JmillsExpensify @saracouto 
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021864619157987293637</li>
        <li>Upwork Job ID: 1864619157987293637</li>
        <li>Last Price Increase: 2024-12-19</li>
<li>Automatic offers: </li>
<ul>
<li>shubham1206agra | Reviewer | 105467437</li>
<li>Krishna2323 | Contributor | 105467438</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @MitchExpensify</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @MitchExpensify</details>

**Comment by JmillsExpensify** [comment]
Added the labels to get this one going.

**Comment by trjExpensify** [comment]
>@dubielzyk-expensify, these messages are also shown in similar way, do we want to update these too?

I'd say yes, such that the design is consistent across them all.

**Comment by dannymcclain** [comment]
@dubielzyk-expensify We already have a different `Receipt Slash` icon in the system, do we want to go with that one or replace it with the one you've made? I don't really think we need two.

![CleanShot 2024-12-05 at 08 08 40@2x](https://github.com/user-attachments/assets/0c010356-a192-4b14-8bee-5c5859ed9ccb)

**Comment by shawnborton** [comment]
Oh good catch Danny, definitely agree we should consolidate.

**Comment by dubielzyk-expensify** [comment]
Yeah that sounds good Danny! Missed that one

**Comment by dannymcclain** [comment]
Right on!

Ok so for the deleted expense version we can just use this existing icon: `App/https://github.com/Expensify/App/blob/main/assets/images/receipt-slash.svg`

**Comment by MitchExpensify** [comment]
It sounds like we're aligned; @dubielzyk-expensify, can you update the OP so we can label this "help wanted," please?

**Comment by dubielzyk-expensify** [comment]
What do you mean "update the OP so we can label this help wanted"?

**Comment by MitchExpensify** [comment]
@dubielzyk-expensify We [made a decision](https://github.com/Expensify/App/issues/53617#issuecomment-2523335851) "for the deleted expense version we can just use this existing icon: App/assets/images/receipt-slash.svg", and I'm wondering if the [original post](https://github.com/Expensify/App/issues/53617#issue-2719271660) (OP) needs to be updated to reflect that decision before we label the issue "help wanted" as a next step

**Comment by dubielzyk-expensify** [comment]
Ahh, I've updated it already 👍

**Comment by MitchExpensify** [comment]
Perfect @dubielzyk-expensify ! 🙇

**Comment by dannymcclain** [comment]
>Waiting for updated UI.

What updates are we waiting on here?

**Comment by MitchExpensify** [comment]
Friendly bump @shubham1206agra:

> What updates are we waiting on here?

**Comment by MitchExpensify** [comment]
@Krishna2323 What is the latest on your PR?

**Comment by rlinoz** [comment]
Hey @Krishna2323 when can we expect a PR?

**Comment by rlinoz** [comment]
@Krishna2323 to get the Deleted expense message
- Create a report with 2 expenses
- Leave a comment in the one you are gonna delete
- Then delete it

<img width="1355" alt="image" src="https://github.com/user-attachments/assets/18950957-9381-415a-8b7e-d83499c44bd4" />

**Comment by rlinoz** [comment]
`Reversed transaction` is gonna be easier to mock, because I think this requires a purchase on the Expensify Card, so you will have to set `isReversedTransaction: true` in a report action that is an expense

I am not sure if we ever show `Deleted report` anymore because of this lines https://github.com/Expensify/App/blob/1cfcfa47ca046760f363114c10b1d77999147f9b/src/libs/ReportUtils.ts#L4077, since the parent action will be deleted we will always show it as `Delete message`, not sure whether that is a bug

**Comment by dubielzyk-expensify** [comment]
Agree that we should probably make it for all messages. I like the idea of using trash as a fallback as well for those instances where we don't have a slash. I'm happy to make custom slash icons though as well. Let's use trash for now and I'll get started on the / icons in the meantime

> Also, separately, I think the idea was to ditch the brackets around the message, so instead of [Deleted message] it would just be Deleted message. That sound right to you all?

Yes agree. It shouldn't have brackets.

**Comment by shawnborton** [comment]
I don't feel strongly there but I think I would say yes for consistency's sake.

**Comment by dubielzyk-expensify** [comment]
Agree with the header treatment as well:

> I'm also down for custom icons but I do kinda agree with Shawn that most people probably won't pick up on the differences. Trash can just says "deleted" so quickly that I'd be down to use it for everything as well. No strong preference!

I did this [originally in my proposal](https://expensify.slack.com/archives/C03U7DCU4/p1733267108521219?thread_ts=1733266656.898339&cid=C03U7DCU4), but most voted for a custom icon. Though I suspect that was before we knew about all the multiple types. 

I'm in favor of using the trash icon on everything:
![CleanShot 2025-01-08 at 14 46 10@2x](https://github.com/user-attachments/assets/89b56e22-bcfc-46cc-a681-c78f3592553e)

**Comment by shawnborton** [comment]
Nice, I say let's do trash can for everything then!

**Comment by dannymcclain** [comment]
Right on—trash it up!

**Comment by dannymcclain** [comment]
What's the difference between a `Hidden message` and a `Deleted message`?

Also, @Expensify/design, how do you feel about the `Reversed transaction` one? I think it makes perfect sense to use the trash icon for anything deleted, but I'm a little skeptical about hidden and reversed. Thoughts?

**Comment by shawnborton** [comment]
Whoa, I've never even heard of a reversed transaction. What does that mean and how does it happen? 

Same comment about hidden message.... how does that happen? I agree with you though Danny, I think we should use some kind of eye with a slash icon for that one.

**Comment by trjExpensify** [comment]
I believe that's a pending Expensify Card transaction that's reversed (like a deposit hold or something) and therefore gets deleted. If there's user comments in that thread to preserve it has the title "[Reversed transaction]".

**Comment by rlinoz** [comment]
Apparently the hidden message is shown when a decision is pending from the moderator.

Briefly looking at the code it seems that a reversed transaction is when you cancel an E-Card transaction

**Comment by shawnborton** [comment]
Okay cool, I think it makes sense to use different icons for those... 

Reversed could use some kind of arrows icon (inspired by Uno?!)
![CleanShot 2025-01-09 at 15 58 34@2x](https://github.com/user-attachments/assets/fdb0d504-374c-47f0-bd98-5c990dacbf25)

Hidden message could use the eye slash icon:
![CleanShot 2025-01-09 at 15 59 14@2x](https://github.com/user-attachments/assets/87c65847-5e2b-4281-a56c-504dc41df0ec)

Thoughts?

**Comment by dannymcclain** [comment]
Yup that's exactly where my mind went as well Shawn. Those look good to me!

**Comment by dannymcclain** [comment]
Looks like that one might not be in the repo yet—here's the SVG!

[Arrows Left Right.svg.zip](https://github.com/user-attachments/files/18367655/Arrows.Left.Right.svg.zip)

**Comment by dubielzyk-expensify** [comment]
Love these additions! Thanks Danny and Shawn!

**Comment by dubielzyk-expensify** [comment]
That looks good to me! 🙌

**Comment by shawnborton** [comment]
Looks good to me too. 

@Krishna2323 in the future, do you mind keeping the conversation consolidated on the PR itself? This way we aren't bouncing back and forth between the issue and PR when reviewing screenshots and leaving comments. Thanks!

**Comment by MitchExpensify** [comment]
*Payment summary*:

- $250 @shubham1206agra requires payment automatic offer (Reviewer)
- $250 @Krishna2323 requires payment automatic offer (Contributor)

**Comment by MitchExpensify** [comment]
@shubham1206agra Can you please complete the Bug Zero checklist here? https://stackoverflow.com/c/expensify/questions/14574

**Comment by MitchExpensify** [comment]
@Krishna2323 Paid and contract ended 👍

**Comment by MitchExpensify** [comment]
Paid and contract ended
